app/models/manageiq/providers/amazon/inventory/persister/cloud_manager.rb

```diff
@@ -1,31 +1,40 @@
 class ManageIQ::Providers::Amazon::Inventory::Persister::CloudManager < ManageIQ::Providers::Amazon::Inventory::Persister
+  include ManageIQ::Providers::Amazon::Inventory::Persister::Shared::CloudCollections
+
   def initialize_inventory_collections
     initialize_tag_mapper
 
-    add_inventory_collections(
-      cloud,
-      %i(vms miq_templates hardwares operating_systems networks disks availability_zones
-         vm_and_template_labels vm_and_template_taggings
-         flavors key_pairs orchestration_stacks orchestration_stacks_resources
-         orchestration_stacks_outputs orchestration_stacks_parameters orchestration_templates)
-    )
-
-    add_inventory_collection(
-      cloud.vm_and_miq_template_ancestry(
-        :dependency_attributes => {
-          :vms           => [collections[:vms]],
-          :miq_templates => [collections[:miq_templates]]
-        }
-      )
-    )
-
-    add_inventory_collection(
-      cloud.orchestration_stack_ancestry(
-        :dependency_attributes => {
-          :orchestration_stacks           => [collections[:orchestration_stacks]],
-          :orchestration_stacks_resources => [collections[:orchestration_stacks_resources]]
-        }
-      )
-    )
+    %i(vms
```

Comment:
@agrare @kbrock lets probably start the review from here. It's a half way between what the old compact way is, vs. the verbose way Marcel did. I think I like this. What do you think?

Reply:
One difference with Marcel's way, this is not class definition. I wonder if we need it on class. While it looks a bit better, it requires a lot of lambdas to pass the dynamic params. What do you think?